### DISCUSSION TITLE
축약형 이름 설계

### DISCUSSION THREAD

**Discussion by coalong** [cowaine/is-my-code-that-strange#99] [Chapter 09-10 (악마, 이름 설계)] [answered]
Java 에서 클래스는 파스칼케이스, 메서드는 카멜케이스를 사용하고 있습니다.

책에서는 기본적으로 이름을 축약하지 않는 것을 권장하지만 축약해야될 경우가 있을 것입니다.

예시로 DTO 나 @firewoody237 님이 올려주신 내용같이 긴 회사명 (NHN, KAC ... 등) 을 사용할 때 축약형 이름은 어떤 네이밍 규칙으로 작성해야할까요?

xxxDTO 이렇게 모두 대문자로 작성해야할지, xxxDto 이렇게 명명 케이스에 맞게 작성해야 될지 궁금합니다.
ex) NHN or Nhn

**Comment by Irisation23** [comment, marked as answer]
실제로 조금 고민을 했었던 기억이 있는 축약어 내용이네요.

NHNResponseDto로 설계할 것인지 NhnResponseDto로 설계할 것인지는 설계한 코드가 구현되기전에 팀원들끼리 맞추고 시작하는 것이 좋아보입니다.
만약 이미 진행되어 누구는 NHNResponseDto로 누구는 NhnResponseDto로 작성하고 있다면, 상급자가 작성한 코드에 맞출 것 같습니다.

그리고 해당 내용에 대한 고민은 이미 외국인들도 고민을 했던 내용 같네요 ㅎㅎ

> 참조
> https://stackoverflow.com/questions/2236807/java-naming-convention-with-acronyms
